src/peft/utils/safelora.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Reference paper: https://arxiv.org/abs/2405.16833
+
+
+import copy
+import os
+from dataclasses import dataclass, field
+from typing import Literal
+
+import torch
+from safetensors import safe_open
+from safetensors.torch import save_file
+
+import peft
+from peft import PeftConfig
+
+from .loftq_utils import _SafetensorLoader as SafetensorLoader
+
+
+@dataclass
+class SafeLoraConfig:
+    """
+    This is the configuration class to store the configuration of a [`safeLora`].
```

Comment:
Regarding [`safeLora`]: There is no object of this name, right? Therefore, this syntax will not work. We could rephrase this to:

> This is the configuration class to set the parameters for SafeLoRA.

Then add a link to the paper.